From 0ceaab39518eb985473b5f09bfb45c56a6a5e9a7 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Thu, 26 Apr 2018 16:43:07 +0300
Subject: [PATCH 3649/5242] MLK-18164: drm: imx: dcss: fix max upscale ratio

commit  ecde7e1640b2449bf9ffd572b25f9ec9c877c2e5 from
https://source.codeaurora.org/external/imx/linux-imx.git

Currently, the maximum upscale ratio is 1:7. However, DCSS can support
upscale ratios up to 1:16, even though the RM states the maximum upscale
ratio is 1:8.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
(cherry picked from commit 361a057ceec0676be069b2150ee533b7ad11227a)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-scaler.c |   12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/imx/dcss/dcss-scaler.c b/drivers/gpu/imx/dcss/dcss-scaler.c
index 48242a5..ee9f69d 100644
--- a/drivers/gpu/imx/dcss/dcss-scaler.c
+++ b/drivers/gpu/imx/dcss/dcss-scaler.c
@@ -521,15 +521,15 @@ struct dcss_scaler_ratios {
 };
 
 static const struct dcss_scaler_ratios dcss_scaler_ratios[] = {
-	{max_downscale(3), max_upscale(5)},
-	{max_downscale(5), max_upscale(7)},
-	{max_downscale(5), max_upscale(7)},
+	{max_downscale(3), max_upscale(16)},
+	{max_downscale(5), max_upscale(16)},
+	{max_downscale(5), max_upscale(16)},
 };
 
 static const struct dcss_scaler_ratios dcss_scaler_wrscl_ratios[] = {
-	{max_downscale(5), max_upscale(5)},
-	{max_downscale(7), max_upscale(7)},
-	{max_downscale(7), max_upscale(7)},
+	{max_downscale(5), max_upscale(16)},
+	{max_downscale(7), max_upscale(16)},
+	{max_downscale(7), max_upscale(16)},
 };
 
 static bool dcss_scaler_fractions_set(struct dcss_soc *dcss, int ch_num,
-- 
1.7.9.5

